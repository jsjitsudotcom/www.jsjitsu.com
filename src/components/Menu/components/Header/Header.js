import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Style from "./Header.scss";
import Menu from "./../../../../assets/material-icons/menu-white.svg";
import Close from "./../../../../assets/material-icons/close-white.svg";
import Ripple from "./../../../Ripple/Ripple";

export default class Header extends PureComponent {
  render() {
    return (
      <header className={Style.container}>
        <Ripple className={Style.menu} onClick={this.props.onMenu}>
          {!this.props.isMenuOpen && <img src={Menu} alt="menu" />}
          {this.props.isMenuOpen && <img src={Close} alt="menu" />}
        </Ripple>

        <div className={Style.title}>{this.props.title}</div>
      </header>
    );
  }
}

Header.propTypes = {
  onClickMenu: PropTypes.func,
  onClickSubmitFeed: PropTypes.func,
  fixed: PropTypes.bool
};
