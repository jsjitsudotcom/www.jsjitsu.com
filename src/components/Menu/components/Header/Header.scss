:local(.container) {
  width: 100%;
  height: 55px;
  background-color: #2d2d32;
  display: flex;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
  backface-visibility: hidden;
  box-shadow: 0 4px 10px rgba(black, 0.2);
}

:local(.menu) {
  flex-shrink: 0;
  position: relative;
  z-index: 2;
  cursor: pointer;
  width: 45px;
  height: 45px;
  display: flex;
  align-items: center;
  border-radius: 4px;
  justify-content: center;
}

:local(.title) {
  font-size: 20px;
  margin-left: 20px;
  color: white;
}
