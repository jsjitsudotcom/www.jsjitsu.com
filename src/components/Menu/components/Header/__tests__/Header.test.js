import React from "react";
import { shallow, mount } from "enzyme";
import { expect as chai } from "chai";
import sinon from "sinon";
import Header from "./../Header";

describe("<Header />", () => {
  it("Le componsant doit être rendu", () => {
    shallow(<Header />);
  });

  it("Vérification que toutes les props soient bien dans les propTypes", () => {
    const props = ["onClickMenu", "onClickSubmitFeed", "fixed"];
    const propTypes = Object.keys(Header.propTypes);

    chai(props).to.deep.eq(propTypes);
  });

  it("Doit afficher le titre et les tabs doivent être fermée", () => {
    const wrapper = shallow(<Header title="Hi" isMenuOpen={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Doit afficher le titre et les tabs doivent être ouverte", () => {
    const wrapper = shallow(<Header title="Hello" isMenuOpen={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Doit appeller la méthode onClick", () => {
    const spy = sinon.spy();
    const wrapper = shallow(<Header onMenu={spy} />);

    wrapper.find("Ripple").prop("onClick")();

    const data = spy.called;
    const expected = true;
    chai(data).to.eq(expected);
  });
});
