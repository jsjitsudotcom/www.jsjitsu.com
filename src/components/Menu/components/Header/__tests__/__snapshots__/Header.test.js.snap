// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Doit afficher le titre et les tabs doivent être fermée 1`] = `
<header
  className="container"
>
  <Ripple
    className="menu"
    color="rgba(255,255,255, 0.5)"
    during={1000}
    onClick={[Function]}
  >
    <img
      alt="menu"
      src="menu-white.svg"
    />
  </Ripple>
  <div
    className="title"
  >
    Hi
  </div>
</header>
`;

exports[`<Header /> Doit afficher le titre et les tabs doivent être ouverte 1`] = `
<header
  className="container"
>
  <Ripple
    className="menu"
    color="rgba(255,255,255, 0.5)"
    during={1000}
    onClick={[Function]}
  >
    <img
      alt="menu"
      src="close-white.svg"
    />
  </Ripple>
  <div
    className="title"
  >
    Hello
  </div>
</header>
`;
